scripts/deduplicate.mjs

```diff
@@ -1,10 +1,11 @@
 /* eslint-disable no-console */
-const { spawn } = require('child_process');
-const path = require('path');
-const fs = require('fs');
-const deduplicate = require('yarn-deduplicate');
+import { spawn } from 'child_process';
+import path from 'path';
+import fs from 'fs';
+import deduplicate from 'yarn-deduplicate';
+import { getWorkspaceRoot } from './utils.mjs';
 
-const lockFile = path.resolve(__dirname, '../yarn.lock');
+const lockFile = path.resolve(getWorkspaceRoot(), 'yarn.lock');
```

Comment:
I see that both `mui-x` and `mui-toolpad` are using this script by simply just re-exporting the file in their respective codebases.
IMO—this causes an issue where this script works correctly only for `material-ui` (original) repository as this is the result of logging out the `lockFile` with current implementation:
```
/Users/lukas/Documents/dev/mui/mui-x/node_modules/@mui/monorepo/yarn.lock
```
I feel that we should op for using `process.cwd()` like in other scripts in this case to produce the following result:
```
/Users/lukas/Documents/dev/mui/mui-x/yarn.lock
```
Or am I missing something (the reason of this script)? 🤔 
P.S. It is not working correctly currently as well. 💡

Reply:
It didn't. Always checked `monorepo` for duplicates. 🙈

Reply:
I never saw this script do anything on toolpad neither. Never investigated why, but this makes sense. If it were up to me, I think I'd prefer to not use `process.cwd()` but instead letting me specify the directory. e.g.

```tsx
import { cli } from '@mui/monorepo/scripts/deduplicate.mjs';
import getProjectRoot from './getProjectRoot.mjs'
cli({ projectRoot: getProjectRoot() })
```

Reply:
I agree. But we could do both—allow for extension and have a sensible default. 🤔 
```mjs
export function cli({ projectRoot } = {}) {

  const lockFile = path.resolve(projectRoot || process.cwd(), 'yarn.lock');
  const yarnlock = fs.readFileSync(lockFile, 'utf8');

  // we could also include some logging to make such issues easier to detect in the future
  console.log(`Looking for duplicates in: "${lockFile}"`);

```

Reply:
I opened a separate issue to discuss this problem: https://github.com/mui/material-ui/issues/35070
Since the scripts did not work previously, this PR doesn't break them. I think we can merge this PR and discuss the next steps in a separate one.